### PR TITLE
BigQuery testing suite that runs against BQ's day 0 region

### PR DESCRIPTION
**Pull request by ahmedabu98**
Creating a new test suite for BigQuery integration tests in region `us-east7` (BQ's early rollout region) so that we can test against the most updated BigQuery back-end. This is so that we can catch new BQ release problems early on and with more clarity (if these tests fail while tests against `us` multi-region pass, it may point to a breaking change in the BQ back-end)

### CODE REVIEW

**Comment by ahmedabu98** [comment]
Manual workflow run on my fork passed successfully: https://github.com/ahmedabu98/beam/actions/runs/6353604637

**Comment by ahmedabu98** [comment]
R: @johnjcasey 
R: @damccorm

**Review by damccorm — APPROVED** [review]
This LGTM from an actions standpoint, I did not review the rest and will defer that to John

**Review by johnjcasey — CHANGES_REQUESTED** [review]

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/testing/BigqueryClient.java [hunk 1]
```diff
@@ -292,6 +292,21 @@ private QueryResponse getTypedTableRows(QueryResponse response) {
   public List<TableRow> queryUnflattened(
       String query, String projectId, boolean typed, boolean useStandardSql)
       throws IOException, InterruptedException {
+    return queryUnflattened(query, projectId, typed, useStandardSql, null);
+  }
+
+  /**
+   * Performs a query without flattening results. May choose a location to perform this operation
```
[the remark this answers is not in the document]
Comment (ahmedabu98) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/testing/BigqueryClient.java:299]: aka GCP region. I'll add that in the doc comments

File: sdks/java/io/google-cloud-platform/build.gradle [hunk 2]
```diff
@@ -244,6 +242,45 @@ task integrationTestKms(type: Test) {
   }
 }
 
+/*
+  Integration tests for BigQueryIO that run on BigQuery's early rollout region (us-east7)
+  with the intended purpose of catching breaking changes from new BigQuery releases.
+  If these tests fail here but not in `Java_GCP_IO_Direct`, there may be a new BigQuery change
+  that is breaking the connector.
```
[the remark this answers is not in the document]
Comment (ahmedabu98) [reply sdks/java/io/google-cloud-platform/build.gradle]: Done

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOStorageReadTableRowIT.java [hunk 3]
```diff
@@ -52,7 +52,13 @@
 @RunWith(JUnit4.class)
 public class BigQueryIOStorageReadTableRowIT {
 
-  private static final String DATASET_ID = "big_query_import_export";
+  private static final String DATASET_ID =
+      TestPipeline.testingPipelineOptions()
+              .as(TestBigQueryOptions.class)
+              .getBigQueryLocation()
+              .equals("us-east7")
```
[the remark this answers is not in the document]
Comment (ahmedabu98) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOStorageReadTableRowIT.java]: Done

**Comment by ahmedabu98** [comment]
Run PostCommit_Java_Dataflow

**Comment by ahmedabu98** [comment]
Run PostCommit_Java_DataflowV2

**Review by johnjcasey — APPROVED** [review]

**Comment by ahmedabu98** [comment]
Run PostCommit_Java_DataflowV2

**Pull request merged**